js/ui/scheduler/workspaces/ui.scheduler.virtual_scrolling.js

```diff
@@ -70,13 +77,38 @@ export default class VirtualScrollingDispatcher {
 
     get scrollingState() {
         return {
-            vertical: this.verticalVirtualScrolling.state,
-            horizontal: this.horizontalVirtualScrolling.state
+            vertical: this.verticalVirtualScrolling?.state,
+            horizontal: this.horizontalVirtualScrolling?.state
         };
     }
     get verticalScrollingState() { return this.scrollingState.vertical; }
     get horizontalScrollingState() { return this.scrollingState.horizontal; }
 
+    get renderState() {
+        const verticalRenderState = this.verticalVirtualScrolling?.getRenderState() || {};
+        const horizontalRenderState = this.horizontalVirtualScrolling?.getRenderState() || {};
+
+        return {
+            ...verticalRenderState,
+            ...horizontalRenderState
+        };
+    }
+
+    get scrollingType() {
+        return this.workspace.option('scrolling.type') ||
+            DefaultScrollingType;
+    }
+
+    get verticalScrollingSupported() {
```

Comment:
I would suggest to rename it, for example, to `verticalScrollingAllowed` - Scheduler supports both types of virtual scrolling no matter what properties a user has provided. But one of the types may be not allowed by the user